apps/chat/agent/tools.py

```diff
@@ -125,21 +123,28 @@ def _move_datetime_to_new_weekday_and_time(date: datetime, new_weekday: int, new
     return date.replace(hour=new_hour, minute=new_minute, second=0) + timedelta(days=day_diff)
 
 
-def create_periodic_task(experiment_session: ExperimentSession, message: str, **kwargs):
-    task_kwargs = json.dumps(
-        {
-            "chat_ids": [experiment_session.participant.identifier],
-            "message": message,
-            "is_bot_instruction": False,
-            "experiment_public_id": str(experiment_session.experiment.public_id),
-        }
-    )
-    PeriodicTask.objects.create(
-        name=f"reminder-{experiment_session.id}-{uuid.uuid4()}",
-        task=BOT_MESSAGE_FOR_USER_TASK,
-        kwargs=task_kwargs,
-        **kwargs,
-    )
+def create_schedule_message(experiment_session: ExperimentSession, message: str, **kwargs):
+    form = ScheduledMessageConfigForm(data=kwargs, experiment_id=experiment_session.experiment.public_id)
+    if form.is_valid():
+        cleaned_data = form.cleaned_data
+        try:
+            with transaction.atomic():
+                experiment = Experiment.objects.get(id=cleaned_data["experiment_id"])
```

Comment:
nit: you can use the experiment from the session `experiment_session.experiment`